### ISSUE TITLE
[ComposedModal] Inner content container missing role

### ISSUE THREAD

**Issue by kennylam** [carbon-design-system/carbon#6122]
## Detailed description
The inner content container of `ComposedModal` is missing `role="dialog`. This can prevent assistive technology from identifying the modal content as separated from the rest of the page.

> What version of the Carbon Design System are you using?
7.11.3

> What did you expect to happen?
Assistive technology, such as screen readers, to identify the modal content container as `role="dialog"`.

> What happened instead?
Attribute `role` is missing from content container element.

> What WCAG 2.1 checkpoint does the issue violate?
https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_overview#focus-order
